app/js/reducers/artifacts.js

```diff
@@ -15,16 +13,12 @@ const initialState = {
 
 const artifacts = (state = initialState, action) => {
     switch (action.type) {
-    case 'ADD_ARTIFACT': {
-        const nextState = ImmutableMap({
-            name: action.artifact.name,
-            uuid: action.artifact.uuid,
-            type: action.artifact.type
-        });
-        return state.set(action.artifact.id, nextState);
+    case 'DELETE_ARTIFACT': {
+        // TODO, make it actually do something
+        const newState = Object.assign({}, state);
```

Comment:
I think it would have, but now as an Array instead of Object it uses `filter()` by index to remove it.